src/js/providers/tracks-mixin.js

```diff
@@ -102,7 +102,11 @@ define(['../utils/underscore',
         if (this._renderNatively) {
             // Only bind and set this.textTrackChangeHandler once so that removeEventListener works
             this.textTrackChangeHandler = this.textTrackChangeHandler || textTrackChangeHandler.bind(this);
-            this.addTracksListener(this.video.textTracks, 'change', this.textTrackChangeHandler);
+
+            this.removeTracksListener(this.video.textTracks, 'change', this.textTrackChangeHandler);
+            if (!this.instreamMode) {
+                this.addTracksListener(this.video.textTracks, 'change', this.textTrackChangeHandler);
+            }
 
             if (utils.isEdge()) {
```

Comment:
@egreaves I know this is something you just added to fix another bug, but does it need a `removeTracksListener`, and should it also check `this.instreamMode`?